src/server/plugins/engine/components/TelephoneNumberField.ts

```diff
@@ -9,8 +9,8 @@ import {
   type FormSubmissionErrors
 } from '~/src/server/plugins/engine/types.js'
 
-const PATTERN = /^[0-9\\\s+()-]*$/
-const DEFAULT_MESSAGE = 'Enter a telephone number in the correct format'
+const PATTERN =
+  /\s*(([+](\s?\d)([-\s]?\d)|0)?(\s?\d)([-\s]?\d){9}|[(](\s?\d)([-\s]?\d)+\s*[)]([-\s]?\d)+)\s*/
```

Comment:
@davidjamesstone Are phone numbers without spaces still valid?

We should differentiate between:

1. Values users can enter as they prefer via `formSchema`
2. Values that are formatted for state via `stateSchema`

From the [guidance for **Phone numbers**](https://design-system.service.gov.uk/patterns/phone-numbers/) it says:

>**Allow different formats**
>
>Let users enter phone numbers in whatever format is familiar to them. Allow for additional spaces, hyphens, dashes and brackets, and be able to accommodate country and area codes.

Reply:
The new regex is liberal and allows users to enter in whatever format they like (see PR description for examples), including brackets and dashes etc.

It just improves it to not match on things like this that were in the tests:

``` 
expect(schema.validate('+111-111-11').error).toBeUndefined()
expect(schema.validate('+111 111 11').error).toBeUndefined()
expect(schema.validate('+11111111').error).toBeUndefined()
```

Reply:
> @davidjamesstone Are phone numbers without spaces still valid?

It looks like it, but can definitely still place spaces where you like?

But from the guidance extract above, maybe this style needs some work?

> (+44) 07777 111222
> +44 (0)7777 111222

Reply:
They're both acceptable. Do you want them adding to the tests?

Reply:
As are variants of the same e.g.

(+44) 07777 111 222
(+44) 07777 111-222
(+44) 07777-111222
(+44) 07777-111-222

etc...

Reply:
Might as well, no harm adding them. Thanks for confirming